x-pack/metricbeat/module/meraki/device_health/device_cellular_gateway_uplink_status.go

```diff
@@ -0,0 +1,58 @@
+package device_health
+
+import (
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	meraki_api "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+func reportCellularGatewayApplianceUplinkStatuses(reporter mb.ReporterV2, organizationID string, devices map[Serial]*Device, responseCellularGatewayUplinkStatuses *meraki_api.ResponseCellularGatewayGetOrganizationCellularGatewayUplinkStatuses) {
+
+	metrics := []mapstr.M{}
+
+	for _, uplink := range *responseCellularGatewayUplinkStatuses {
+
+		if device, ok := devices[Serial(uplink.Serial)]; ok {
+			metric := mapstr.M{
+				"cellular.gateway.uplink.network_id":       uplink.NetworkID,
+				"cellular.gateway.uplink.last_reported_at": uplink.LastReportedAt,
+				"device.address":                           device.Address,
+				"device.firmware":                          device.Firmware,
+				"device.imei":                              device.Imei,
+				"device.lan_ip":                            device.LanIP,
+				"device.location":                          device.Location,
+				"device.mac":                               device.Mac,
+				"device.model":                             device.Model,
+				"device.name":                              device.Name,
+				"device.network_id":                        device.NetworkID,
+				"device.notes":                             device.Notes,
+				"device.product_type":                      device.ProductType,
+				"device.serial":                            device.Serial,
+				"device.tags":                              device.Tags,
+			}
+
+			for _, item := range *uplink.Uplinks {
+				metrics = append(metrics, mapstr.Union(metric, mapstr.M{
+					"cellular.gateway.uplink.apn":              item.Apn,
```

Comment:
do these need to be named differently from other uplink fields, or are the MG uplinks a distinct concept? 

AFAICT the MG uplink metadata is just a superset of the other uplink fields (except 'ip_assigned_by')

```
"uplink.interface"
"uplink.status"
"uplink.ip"
"uplink.gateway"
"uplink.public_ip"
"uplink.primary_dns"
"uplink.secondary_dns"
"uplink.ip_assigned_by"

------------------

"cellular.gateway.uplink.interface"
"cellular.gateway.uplink.status"
"cellular.gateway.uplink.ip"
"cellular.gateway.uplink.gateway"
"cellular.gateway.uplink.public_ip"
"cellular.gateway.uplink.dns1"
"cellular.gateway.uplink.dns2"

"cellular.gateway.uplink.apn"
"cellular.gateway.uplink.connection_type"
"cellular.gateway.uplink.iccid"
"cellular.gateway.uplink.model"
"cellular.gateway.uplink.provider"
"cellular.gateway.uplink.signal_stat.rsrp"
"cellular.gateway.uplink.signal_stat.rsrq"
"cellular.gateway.uplink.signal_type"
```

Reply:
Fair point, I agree the data looks the same.  However, I have already seen where meraki as two unique api calls and same ip address is not the same. Also in this case it does not appear to be a unique meraki API call ... client.Appliance.GetOrganizationApplianceUplinkStatuses() and client.CellularGateway.GetOrganizationCellularGatewayUplinkStatuses()  ... unless I seen the data side by side and returning the exact same data, I am not sure I feel comfortable, assuming the APIs are returning the same values.  If I return what Meraki returns and do not try to merge / combine it, then if there is an issue it is there issue and not my issue.   For the naming pattern ... I was trying to do Object,  "cellulargateway" in naming for future debug.